lib/utils/aws/signing.go

```diff
@@ -76,8 +71,8 @@ func (s *SigningServiceConfig) CheckAndSetDefaults() error {
 		}
 		s.Session = ses
 	}
-	if s.GetSigningCredentials == nil {
-		s.GetSigningCredentials = GetAWSCredentialsFromSTSAPI
+	if s.CredentialsGetter == nil {
+		s.CredentialsGetter = NewCredentialsGetter()
```

Comment:
Should we use a caching one here?

Reply:
I've changed this to use the `cachedCredentialsGetter` by default.

However, the database side (dynamodb/opensearch) creates a new signing service per connection (so a new cache every connection).

@Tener @smallinsky do you guys think it's worth using a cache for database access as well? If so, I can make a separate change to pass a cached getter to `common.EngineConfig`

Reply:
Yes, If we are sure that this won't break anything I think that is reasonable to use cached getter.

Reply:
I tried this (fe0fc168b413321debd0afbca7344feb8191c0f3) but it does not work for database access because the expiry changes every time.

I believe the cause is auth signs a new cert with a fixed TTL on every new database connection:
https://github.com/gravitational/teleport/blob/e461f29554291465a335d542f06a2de75d470ec2/lib/srv/db/proxyserver.go#L615-L622

The identity.Expiry db service receives is never the real expiry on the client side.